src/error.rs

```diff
@@ -477,6 +479,18 @@ impl From<http::method::InvalidMethod> for Parse {
     }
 }
 
+impl From<http::header::InvalidHeaderName> for Parse {
```

Comment:
I'd say these `From` implementations are perhaps a little too implicit. I think we should likely be sure to be explicit when labeling that an error in an internal bug.

Reply:
I agree they are implicit, i initially did not have it but then the lint complains the _variant is never constructed_ (even though it is) for the Parse::Internal error

Reply:
That's ok, it's probably because the variants are only "used" in release mode. It's possible to add an `#[cfg_attr(debug_assertions, allow(unused))]` on the variant. Then we don't need these impls.

Reply:
Sweet, i wasn't sure if supressing the lint was the right approach. Thanks for the help!